src/view/statistics.js

```diff
@@ -0,0 +1,367 @@
+import Chart from 'chart.js';
+import ChartDataLabels from 'chartjs-plugin-datalabels';
+import { getTimeDifferenceMs, getTimeFormatted } from './utils/point.js';
+import SmartView from './smart.js';
+
+const BAR_HEIGHT = 55;
+const MIN_TYPES_COUNT = 5;
+
+const getTypes = (points) => {
+  const uniqueTypes = new Set();
+
+  points.forEach((currentValue) => {
+    uniqueTypes.add(currentValue.type.toUpperCase());
+  });
+
+  return Array.from(uniqueTypes.values());
+};
+
+const createStatisticsTemplate = (points) => {
+  const typesCount = getTypes(points).length;
+
+  let height = BAR_HEIGHT * MIN_TYPES_COUNT;
+
+  if (typesCount > MIN_TYPES_COUNT) {
+    height = BAR_HEIGHT * typesCount;
+  }
+
+  return `<section class="statistics">
+            <h2 class="visually-hidden">Trip statistics</h2>
+            <div class="statistics__item">
+              <canvas class="statistics__chart" id="money" width="900" height="${height}"></canvas>
+            </div>
+            <div class="statistics__item">
+              <canvas class="statistics__chart" id="type" width="900" height="${height}"></canvas>
+            </div>
+            <div class="statistics__item">
+              <canvas class="statistics__chart" id="time-spend" width="900" height="${height}"></canvas>
+            </div>
+          </section>`;
+};
+
+const getSumPriceOfTypes = (points) => {
+  let uniqueTypes = getTypes(points);
+
+  let sumPrices = uniqueTypes.map((type) => {
+    let sumPrice = 0;
+
+    points.forEach((point) => {
+      if (point.type.toUpperCase() === type) {
+        sumPrice += point.price;
+      }
+    });
```

Comment:
```suggestion
    points.reduce((acc, it) => {
      const isCurrentType = point.type.toUpperCase() === type;
    
      return isCurrentType ? acc += it.price : acc
    }, 0)
```
мы же уже с тобой переделывали форИчи на редьюсы 🤔